src/pfe/file-watcher/server/test/scripts/exec.sh

```diff
@@ -24,29 +24,49 @@ Options:
 EOF
 }
 
+function checkExitCode() {
+	exit_code=$1
+	error_msg=$2
+	if [[ $exit_code -eq 0 ]]; then
+		echo -e "${GREEN}✔ Done. ${RESET}\n"
+	else
+		echo -e "${RED}✖ $error_msg  ${RESET}\n"
+		exit 1
+	fi
+}
+
 function downloadCwctl() {
+    echo -e "${BLUE}>> Downloading latest installer ... ${RESET}"
     EXECUTABLE_NAME="cwctl"
-    DOWNLOAD_CMD="curl -X GET http://download.eclipse.org/codewind/codewind-installer/master/latest/cwctl-linux --output /usr/local/bin/$EXECUTABLE_NAME"
-    if [ $TEST_TYPE == "local" ]; then
-        docker exec -i $CODEWIND_CONTAINER_ID bash -c "$DOWNLOAD_CMD"
-        docker exec -i $CODEWIND_CONTAINER_ID bash -c "chmod +x /usr/local/bin/$EXECUTABLE_NAME"
-    elif [ $TEST_TYPE == "kube" ]; then
-        kubectl exec -i $CODEWIND_POD_ID -- bash -c "$DOWNLOAD_CMD"
-        kubectl exec -i $CODEWIND_POD_ID -- bash -c "chmod +x /usr/local/bin/$EXECUTABLE_NAME"
+    HOST_OS=$(uname -a)
+    if [[ "$HOST_OS" =~ "Darwin" ]]; then
+        extension="macos"
+    elif [[ "$HOST_OS" =~ "Linux" ]]; then
+        extension="linux"
     fi
+    echo "Extension is $extension"
+    curl -X GET http://download.eclipse.org/codewind/codewind-installer/master/latest/cwctl-$extension --output $EXECUTABLE_NAME
```

Comment:
switched back to saving cwctl to pwd?

Reply:
yes because this will now be on the users machine (i.e on the test VM) and we should not be writing and installing cwctl to their /usr/local/bin folders. So better to just do it within our test folder and also gitignored it so it does not get comitted accidentally. We also remove it after the test run.